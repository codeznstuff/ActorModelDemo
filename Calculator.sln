
Microsoft Visual Studio Solution File, Format Version 12.00
# Visual Studio Version 16
VisualStudioVersion = 16.0.29728.190
MinimumVisualStudioVersion = 10.0.40219.1
Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "AkkaDemo", "AkkaDemo\AkkaDemo.csproj", "{9092B239-E488-474C-9F92-97AB22C6FF5B}"
EndProject
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "OrleansDemo", "OrleansDemo\OrleansDemo.csproj", "{601B6D46-532C-4239-8106-532CCD229D3F}"
EndProject
Global
	GlobalSection(SolutionConfigurationPlatforms) = preSolution
		Debug|Any CPU = Debug|Any CPU
		Release|Any CPU = Release|Any CPU
	EndGlobalSection
	GlobalSection(ProjectConfigurationPlatforms) = postSolution
		{9092B239-E488-474C-9F92-97AB22C6FF5B}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{9092B239-E488-474C-9F92-97AB22C6FF5B}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{9092B239-E488-474C-9F92-97AB22C6FF5B}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{9092B239-E488-474C-9F92-97AB22C6FF5B}.Release|Any CPU.Build.0 = Release|Any CPU
		{601B6D46-532C-4239-8106-532CCD229D3F}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{601B6D46-532C-4239-8106-532CCD229D3F}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{601B6D46-532C-4239-8106-532CCD229D3F}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{601B6D46-532C-4239-8106-532CCD229D3F}.Release|Any CPU.Build.0 = Release|Any CPU
	EndGlobalSection
	GlobalSection(SolutionProperties) = preSolution
		HideSolutionNode = FALSE
	EndGlobalSection
	GlobalSection(ExtensibilityGlobals) = postSolution
		SolutionGuid = {6F91C814-7E84-407C-8AC1-2763A566717C}
	EndGlobalSection
EndGlobal
