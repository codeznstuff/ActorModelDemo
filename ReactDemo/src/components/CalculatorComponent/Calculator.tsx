import React, { FC, useLayoutEffect, ReactElement } from 'react';
import { OperationType, OperatorDefinition, ButtonValueMap } from '../../@types/constants';
import styles from './Calculator.module.scss';
import { Button } from '../ButtonComponent/Button';
import { Input } from '../InputComponent/Input';
import { ClearButton } from '../ClearButtonComponent/ClearButton';
import Calc from '../../utils/calculation';
import { useStateContext } from '../../context/context';

const Calculator: FC = () => {
  const { addToInput, previousValue, getOperation, handleOperation, handleClear, handleEqual, currentValue, keyPress } = useStateContext();

  /* Ignore in test coverage report due to simplicity*/
  /* istanbul ignore next */
  const handleKeyPress = (event: KeyboardEvent) => {
    if (!Calc.isNaN(event.key) || event.key === '.') {
      addToInput(event.key);
    } else if (event.key === '+') {
      handleOperation(OperationType.Addition);
    } else if (event.key === '-') {
      handleOperation(OperationType.Subtraction);
    } else if (event.key === '*') {
      handleOperation(OperationType.Multiplication);
    } else if (event.key === '/') {
      handleOperation(OperationType.Division);
    } else if (event.key === 'Enter') {
      handleEqual();
    } else if (event.key === 'c') {
      handleClear();
    } else {
      return;
    }
    keyPress(getButtonId(event.key));
  };
  useLayoutEffect(() => {
    window.addEventListener('keyup', handleKeyPress);
    return () => {
      window.removeEventListener('keyup', handleKeyPress);
    };
  });

  const getButtonId = (key: string): string => {
    let isButtonASymbol = ButtonValueMap.find(x => x.key === key);
    return isButtonASymbol ? isButtonASymbol.value : key;
  };
  const generateButtons = (values: Array<string>, operator: Array<OperatorDefinition>): Array<ReactElement> => {
    let result = [];
    for (var i = 0; i < values.length; i++) {
      let newButton = <Button id={getButtonId(values[i])} key={values[i]} value={values[i]} handleClick={addToInput} />;
      result.push(newButton);
    }
    for (var ii = 0; ii < operator.length; ii++) {
      let o = operator[ii].op;
      result.push(
        <Button
          id={getButtonId(operator[ii].value)}
          key={operator[ii].value}
          value={operator[ii].value}
          operator
          handleClick={() => handleOperation(o)}
        />
      );
    }
    return result;
  };

  return (
    <div className={styles.app} data-testid="calculator">
      <div className={styles.calculator_wrapper}>
        <Input previousInput={previousValue + getOperation()} input={currentValue} />
        <div className={styles.row}>{generateButtons(['7', '8', '9'], [{ value: 'x', op: OperationType.Multiplication }])}</div>
        <div className={styles.row}>{generateButtons(['4', '5', '6'], [{ value: '-', op: OperationType.Subtraction }])}</div>
        <div className={styles.row}>{generateButtons(['1', '2', '3'], [{ value: '+', op: OperationType.Addition }])}</div>
        <div className={styles.row}>
          {generateButtons(
            ['0', '.'],
            [
              { value: '=', op: OperationType.Equal },
              { value: '/', op: OperationType.Division }
            ]
          )}
        </div>
        <div className={styles.row}>
          <ClearButton handleClear={handleClear} />
        </div>
      </div>
    </div>
  );
};

export default Calculator;
