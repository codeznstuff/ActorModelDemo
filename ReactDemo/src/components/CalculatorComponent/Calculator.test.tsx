import React from 'react';
import { cleanup } from '@testing-library/react'
import Calculator from './Calculator';
import Enzyme, { shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

describe("CalculatorComponent", () => {
    let calc: ShallowWrapper;
    beforeEach(() => {
        calc = shallow(<Calculator />);
    });
    it('Calculator Snapshot should match', () => {
        expect(calc.debug()).toMatchSnapshot();
    })

    afterEach(cleanup);
})

