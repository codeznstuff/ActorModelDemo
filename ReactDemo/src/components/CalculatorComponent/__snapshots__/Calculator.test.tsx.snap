// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorComponent Calculator Snapshot should match 1`] = `
"<div className=\\"app\\" data-testid=\\"calculator\\">
  <div className=\\"calculator_wrapper\\">
    <Input previousInput=\\"\\" input=\\"\\" />
    <div className=\\"row\\">
      <Button id=\\"7\\" value=\\"7\\" handleClick={[Function: addToInput]} />
      <Button id=\\"8\\" value=\\"8\\" handleClick={[Function: addToInput]} />
      <Button id=\\"9\\" value=\\"9\\" handleClick={[Function: addToInput]} />
      <Button id=\\"x\\" value=\\"x\\" operator={true} handleClick={[Function: handleClick]} />
    </div>
    <div className=\\"row\\">
      <Button id=\\"4\\" value=\\"4\\" handleClick={[Function: addToInput]} />
      <Button id=\\"5\\" value=\\"5\\" handleClick={[Function: addToInput]} />
      <Button id=\\"6\\" value=\\"6\\" handleClick={[Function: addToInput]} />
      <Button id=\\"Minus\\" value=\\"-\\" operator={true} handleClick={[Function: handleClick]} />
    </div>
    <div className=\\"row\\">
      <Button id=\\"1\\" value=\\"1\\" handleClick={[Function: addToInput]} />
      <Button id=\\"2\\" value=\\"2\\" handleClick={[Function: addToInput]} />
      <Button id=\\"3\\" value=\\"3\\" handleClick={[Function: addToInput]} />
      <Button id=\\"Plus\\" value=\\"+\\" operator={true} handleClick={[Function: handleClick]} />
    </div>
    <div className=\\"row\\">
      <Button id=\\"0\\" value=\\"0\\" handleClick={[Function: addToInput]} />
      <Button id=\\"Period\\" value=\\".\\" handleClick={[Function: addToInput]} />
      <Button id=\\"Enter\\" value=\\"=\\" operator={true} handleClick={[Function: handleClick]} />
      <Button id=\\"Divide\\" value=\\"/\\" operator={true} handleClick={[Function: handleClick]} />
    </div>
    <div className=\\"row\\">
      <ClearButton handleClear={[Function: handleClear]} />
    </div>
  </div>
</div>"
`;
